GoogleForCreators/web-stories-wp
# Dashboard: Add No Results Message To Filters

## Context

As part of the work slated to improve Dashboard filtering, this adds a consistent "Sorry, we couldn't find any results." when a user has searched and used author and/or taxonomy filters, or keyboard search.

✅  Important note: This work relies on the changes in `feature/11620_FiltersProvider_add_authors_filter` as a branch to verify this works as intended. When #11620 has been merged, this can point towards the `main` branch instead of that feature branch.

## Summary

Note: Updated screenshots as of June 22nd, 2022
**Example of keyword search returning no results:**
![Screen Shot 2022-06-22 at 10 10 19 AM](https://user-images.githubusercontent.com/7110244/175050784-4d356d04-7fe2-4bcd-a662-b6a7124341e1.png)

**Example of filters search returning no results:**
![Screen Shot 2022-06-22 at 10 10 27 AM](https://user-images.githubusercontent.com/7110244/175050791-944e0294-dbdf-4734-9e87-b029d3acbccd.png)

**Example of no stories to display with no filters and no keywords enabled:**
![Screen Shot 2022-06-22 at 10 10 34 AM](https://user-images.githubusercontent.com/7110244/175050802-d45ead69-4622-4f8e-94a4-4fd6dee81d90.png)



## Relevant Technical Choices

This imports the `filtersObjects` data and uses it to determine whether a filter has been enabled. If the `filterObjects` is empty, we haven't utilized any of the `FilterProvider` filters. If it contains items, and there are no stories to display, we can safely assume that the `No results` message should display.

The same logic is used for the `Explore Templates` button, which should only display when there are no stories created at all.

## To-do


Nothing.

## User-facing changes


This makes the `no results found` message consistent, regardless of whether a keyword search was used, or if multiple filters have been used.

## Testing Instructions






This PR can be tested by following these steps:

1. Navigate to the Dashboard
2. Use the filters and/or keyword search to return a result that has no matches
3. See the "Sorry, we couldn't find any results." message specific to whether a keyword was used
4. The "Explore Templates" button should only display if there are no stories to view


## Reviews

### Does this PR have a security-related impact?


No.

### Does this PR change what data or activity we track or use?


No.

### Does this PR have a legal-related impact?


No.

## Checklist



- [] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))

---



Fixes #11664
